compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -955,10 +955,14 @@ impl<'a> State<'a> {
                 self.pclose();
             }
             ast::TyKind::AnonymousStruct(ref fields, ..) => {
+                self.cbox(1);
+                self.ibox(7);
                 self.s.word("struct");
                 self.print_record_struct_body(fields, ty.span);
             }
             ast::TyKind::AnonymousUnion(ref fields, ..) => {
+                self.cbox(1);
+                self.ibox(6);
```

Comment:
I suggest using whatever `unsafe { ... }` or `loop { ... }` use instead of trying to pick the numbers manually.

Reply:
I tried to do that but that outputted
```
struct Foo {
    _: union {
        },
}
```
which makes sense, because "union" is the head, however the correct formatting must be
struct Foo {
    _: union {
    },
}
```
I tried to make somehow `_` a head but that also didn't work.

This works as intended, but I'm still going to try some more things, maybe I can do something inside `print_record_struct_body` to make it work.